var/spack/repos/builtin/packages/quantlib/package.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Quantlib(CMakePackage):
+    """The QuantLib project is aimed at providing a comprehensive software
+    framework for quantitative finance. QuantLib is a free/open-source library
+    for modeling, trading, and risk management in real-life."""
+
+    homepage = "https://www.quantlib.org/"
+    url = "https://github.com/lballabio/QuantLib/releases/download/v1.31.1/QuantLib-1.31.1.tar.gz"
+
+    maintainers("TheQueasle")
+
+    version("1.31.1", sha256="13b5346217153ae3c185e0c640cc523a1a6522c3a721698b2c255fd9a1a15a68")
```

Comment:
Checksum verified. By now there is a version 1.34 (didn't check for changed dependencies).
```suggestion
    version("1.34", sha256="eb87aa8ced76550361771e167eba26aace018074ec370f7af49a01aa56b2fe50")
    version("1.31.1", sha256="13b5346217153ae3c185e0c640cc523a1a6522c3a721698b2c255fd9a1a15a68")
```